packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -4,58 +4,192 @@
  *--------------------------------------------------------------------------------------------*/
 import * as React from 'react';
 import cx from 'classnames';
-import { Popover, Box } from '../utils/index.js';
-import type { CommonProps, PopoverProps } from '../utils/index.js';
+import {
+  useMergeRefs,
+  useFloating,
+  autoUpdate,
+  offset,
+  flip,
+  shift,
+  useHover,
+  useFocus,
+  useDismiss,
+  useRole,
+  useInteractions,
+  useClick,
+  FloatingPortal,
+} from '@floating-ui/react';
+import type { Placement } from '@floating-ui/react';
+import { Box, type PolymorphicForwardRefComponent } from '../utils/index.js';
+import { ThemeContext } from '../ThemeProvider/ThemeContext.js';
+import ReactDOM from 'react-dom';
 
-export type TooltipProps = {
+type TooltipOptions = {
+  /**
+   * Placement of the Tooltip
+   * @default 'top'
+   */
+  placement?: Placement;
+  /**
+   * Property for manual visibility control
+   */
+  visible?: boolean;
+  /**
+   * Function that allows users to control Tooltip visibility
+   */
+  toggleVisible?: (open: boolean) => void;
+  /**
+   * Use this if you want Tooltip to follow moving trigger element
+   * @default false;
+   */
+  followTrigger?: boolean;
+};
+
+type TooltipOwnProps = {
   /**
    * Content of the tooltip.
    */
-  content: React.ReactNode;
+  content?: React.ReactNode;
   /**
    * Element to have tooltip on. Has to be a valid JSX element and needs to forward its ref.
    * If not specified, the `reference` prop should be used instead.
    */
-  children?: JSX.Element;
-} & Omit<PopoverProps, 'className'> &
-  Omit<CommonProps, 'title'>;
+  children?: React.ReactNode;
+  /**
+   * Element to append tooltip to.
+   * Appends to ThemeProvider portalContainerRef by default.
+   */
+  appendTo?: HTMLElement;
```

Comment:
renamed to portal. Not adding boolean version for now.

Reply:
why not? that's the whole point of that prop - it will give a simple way to toggle between `appendTo={portalContainer}` and `appendTo='parent'`.

without boolean, there is no easy way to append to parent.